clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1385,7 +1385,7 @@ private void handleOffsetOutOfRange(FetchPosition fetchPosition, TopicPartition
             byte[] valueByteArray = valueBytes == null ? null : Utils.toArray(valueBytes);
             V value = valueBytes == null ? null : this.valueDeserializer.deserialize(partition.topic(), headers, valueByteArray);
             return new ConsumerRecord<>(partition.topic(), partition.partition(), offset,
-                                        timestamp, timestampType, record.checksumOrNull(),
```

Comment:
Could we remove `checksumOrNull` from `Record`? That method works for `AbstractLegacyRecordBatch` only.

Reply:
I had initially removed it. But I added it back since it's used by `DumpLogSegments`.

Reply:
`DumpLogSegments` can call `RecordBatch#checksum` instead of `checksumOrNull` if the record is legacy. I can file a PR as follow-up to address that.

Reply:
I don't think that suggestion works. However, you could add conditional code in `DumpLogSegments` so that the crc at the record level is only printed if it's `AbstractLegacyRecordBatch`. Sounds good as a follow-up.

Reply:
> Sounds good as a follow-up.

#10498